test/python/pulse/compiler_passes/test_set_sequence.py

```diff
@@ -0,0 +1,379 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test SetSequence"""
+from test import QiskitTestCase
+from ddt import ddt, named_data, unpack
+
+from qiskit.pulse import (
+    Constant,
+    Play,
+    Delay,
+    ShiftPhase,
+)
+
+from qiskit.pulse.ir import (
+    SequenceIR,
+)
+
+from qiskit.pulse.model import QubitFrame, Qubit, MixedFrame
+from qiskit.pulse.transforms import (
+    AlignLeft,
+    AlignRight,
+    AlignSequential,
+    AlignFunc,
+    AlignEquispaced,
+)
+from qiskit.pulse.compiler import MapMixedFrame, SetSequence
+from qiskit.pulse.exceptions import PulseError
+from .utils import PulseIrTranspiler
+
+
+@ddt
+class TestSetSequenceParallelAlignment(QiskitTestCase):
```

Comment:
In a typical circuit pass test, we use a public conversion function and test program is provided in a form of circuit, rather than a DAG circuit. Do you want to take the same pattern? I'm also fine with current pattern.
https://github.com/Qiskit/qiskit/blob/8b6e4fe0fe55fd96e182e526860321db702a7a54/test/python/transpiler/test_apply_layout.py#L36-L49

Reply:
I had the tests in terms of IR, so I left it like this. Once the builder is set up for the new model, I can modify the tests to be based on ScheduleBlock.